pkg/controller/util.go

```diff
@@ -26,11 +27,14 @@ import (
 	jsonpatch "github.com/evanphx/json-patch"
 	v1 "k8s.io/api/core/v1"
 	storage "k8s.io/api/storage/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/klog"
 )
 
+var ctx context.Context = context.Background()
```

Comment:
Please do not use global variables. Use `context.TODO()` in all places where you use this.
Or even better, find a good API server timeout and use that.

Reply:
Sure, let me start with context.TODO() local variable. Will do a follow up PR for configurable timeout . thanks !

Reply:
@jsafrane done.. changed to context.TODO() . .. @msau42 @pohly please review!